src/app/joke.ts

```diff
@@ -0,0 +1,15 @@
+export class Joke {
```

Comment:
I know this is the default file/location which is chosen when you use the angular cli to generate a class.

But some food for thought for future projects, it's good to keep related things together. So in this case I might stick the joke.ts in the joke folder, since that is the folder where I store things related to a "singular" joke.

In more complex projects with lots of models i'm of the opinion you should stick them all in a folder called "models", so you know where they all are (that also works better if you start using much more advanced angular architecture packages like ngRx, but thats an advanced topic for another time!.